codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/RecursiveConstraintViolationsTest.kt

```diff
@@ -0,0 +1,185 @@
+/*
```

Comment:
I'm finding this test harness somewhat hard to follow.
I think it'd be easier to decouple the model generation step from the test execution step, persisting (and committing) the generated models with a check to make sure they stay fresh. 

This would require some fiddling with Gradle, which I understand might be unpalatable.

Reply:
Not sure how that can be achieved. These test case generators often produce models coupled to values that are used in the assertions (e.g. `shapeIdWithConstraintViolationRustBoxTrait` here).